worldwind-gt-geopkg/src/main/java/gov/nasa/worldwind/geopkg/mosaic/GeoPackageReader.java

```diff
@@ -90,21 +90,44 @@
     protected File sourceFile;
 
     protected Map<String, TileEntry> tiles = new HashMap<String, TileEntry>();
+    
+    // TODO: store these gridRanges in the TileEntry class
+    protected Map<String, GridEnvelope2D> gridRanges = new HashMap<String, GridEnvelope2D>();
 
     public GeoPackageReader(Object source, Hints hints) throws IOException {
         coverageFactory = CoverageFactoryFinder.getGridCoverageFactory(this.hints);
         sourceFile = GeoPackageFormat.getFileFromSource(source);
         GeoPackage file = new GeoPackage(sourceFile);
 
         try {
-            // Set the coverage name to the name of the the first raster tileset.
             coverageName = null;   // default value was "geotools_coverage"
-            for (TileEntry tile : file.tiles()) {
-                tiles.put(tile.getTableName(), tile);
+            for (TileEntry tileset : file.tiles()) {
+                // Map the tileset to the coverage name (table name)
+                tiles.put(tileset.getTableName(), tileset);
+
+                // Set the default coverage name to the name of the the first raster tileset.
                 if (coverageName == null) {
-                    coverageName = tile.getTableName();
+                    coverageName = tileset.getTableName();
                 }
+
+                // TODO: compute the gridRange in the GeoPackage class
+                List<TileMatrix> matricies = tileset.getTileMatricies();
+                TileMatrix matrix = matricies.get(matricies.size() - 1);
+                int maxZoomLevel = matrix.getZoomLevel();
```

Comment:
To clarify, "coverage tiles table" refers to GeoPackage tile pyramid user data tables?